arch/xtensa/include/esp32/memory_layout.h

```diff
@@ -80,18 +83,16 @@
 #endif
 
 #ifdef CONFIG_XTENSA_IMEM_USE_SEPARATE_HEAP
-#define	XTENSA_IMEM_REGION_SIZE	CONFIG_XTENSA_IMEM_REGION_SIZE
+#  define	XTENSA_IMEM_REGION_SIZE	CONFIG_XTENSA_IMEM_REGION_SIZE
 #else
-#define	XTENSA_IMEM_REGION_SIZE	0
+#  define	XTENSA_IMEM_REGION_SIZE	0
 #endif
 
-/* If CONFIG_XTENSA_IMEM_MAXIMIZE_HEAP_REGION is defined, it means
- * using maximum separate heap for internal memory, but part of
- * the available memory is reserved for the Region 1 heap.
- */
+/* Internal heap starts at the end of the ROM data. */
 
-#ifdef CONFIG_XTENSA_IMEM_MAXIMIZE_HEAP_REGION
-#ifndef HEAP_REGION_OFFSET
-#define HEAP_REGION_OFFSET      0x2000
-#endif
+#ifndef CONFIG_SMP
+#  define ESP32_IMEM_START  0x3ffe1330
```

Comment:
how about:
```
#define ESP32_IMEM_START HEAP_REGION2_START
```

Reply:
It's not always the start of region2, in SMP it's the start of region3.

I can change it to:
```C
#ifndef CONFIG_SMP
#  define ESP32_IMEM_START  HEAP_REGION2_START
#else
#  define ESP32_IMEM_START  HEAP_REGION3_START
#endif
```
Which I think is better.

Reply:
ok. when i made this comment, i was looking at the individual commit https://github.com/apache/incubator-nuttx/pull/3003/commits/7911aa57372e6f6002a2cc98fc5d420d379f968a which didn't have region3 yet.